addon/services/toast.ts

```diff
@@ -0,0 +1,185 @@
+   * @param {Object} opts - Toast options
+   * @param {boolean} [opts.closeButton=true] - Display a button to close the toast manually.
+   */
+  error: ToastFunction = this._delegate(ToastType.ERROR);
+
+  /**
+   * Display a warning toast
+   *
+   * @param {string} message - The Toast's message
+   * @param {string} title - The toast's title
+   * @param {Object} opts - Toast options
+   * @param {boolean} [opts.closeButton=true] - Display a button to close the toast manually.
+   */
+  warning: ToastFunction = this._delegate(ToastType.WARNING);
+
+  /**
+   * Display an information toast
+   *
+   * @param {string} message - The Toast's message
+   * @param {string} title - The toast's title
+   * @param {Object} opts - Toast options
+   * @param {boolean} [opts.closeButton=true] - Display a button to close the toast manually.
+   */
+  info: ToastFunction = this._delegate(ToastType.INFO);
+
+  private _delegate(type: ToastType): ToastFunction {
+    return (message: string, title: string, opts: ToastOptions = DEFAULT_OPTIONS): void => {
+      if (isEmpty(message) && isEmpty(title)) return;
+
+      this._buildToast(type, message, title, { ...DEFAULT_OPTIONS, ...opts });
+    };
+  }
+
+  private _buildToast(type: ToastType, message: string, title: string, opts: ToastOptions): void {
+    const container: Element = this._buildContainer();
+
+    const toast: Element = document.createElement('div');
+    toast.classList.add('toast', 'toast--hidden', `toast-${type}`);
+
+    const titleContainer: Element = this._buildToastPart('div', ['toast-title'], title);
+    const messageContainer: Element = this._buildToastPart('div', ['toast-message'], message);
+
+    let toastChildren: Element[] = [titleContainer, messageContainer];
+
+    if (opts.closeButton) {
+      const closeButton: Element = document.createElement('button');
+
+      closeButton.classList.add('toast-close-button');
+      closeButton.innerHTML = '<i class="fas fa-times text-size-5"></i>';
+      closeButton.addEventListener('click', this._onToastClose.bind(this), { once: true });
+
+      toastChildren.push(closeButton);
+    }
+
+    if (opts.onclick && typeof opts.onclick === 'function') {
+      toast.addEventListener('click', (e) => {
+        e.stopPropagation();
+        opts.onclick && opts.onclick(e);
+      }, { once: true });
+    }
+
+    if (opts.tapToDismiss) {
+      toast.addEventListener('click', () => { this._destroyToast(toast) }, { once: true });
+    }
+
+    toastChildren.forEach((x) => toast.append(x));
+
+    if (container.querySelector('.toast')) {
+      container.insertBefore(toast, container.querySelector('.toast'));
+    } else {
+      container.append(toast);
+    }
+
+    toast.classList.remove('toast--hidden');
+    toast.classList.add('toast--visible');
+
+    this._toasts.push(toast);
+
+    if (opts.timeout === 'none') {
+      return;
+    }
+
+    later(this, () => this._destroyToast(toast), opts.timeout || DEFAULT_TOAST_TIMEOUT);
```

Comment:
Move this logic into a separate method ?
`handleToastDisplay`, `displayToast` ? (idk 🤷 )